semantics/src/namespace/operations.rs

```diff
@@ -33,6 +35,14 @@ fn index_map(map: Map, index: Type) -> Result<Type, SemanticError> {
     Ok(*map.value)
 }
 
+fn index_tuple(_tuple: Tuple, index: Type) -> Result<Type, SemanticError> {
+    if index != Type::Base(Base::U256) {
+        return Err(SemanticError::TypeError);
+    }
+    // Not sure how I would get the type here given the inputs..
```

Comment:
I don't think it should be possible to index tuples with a runtime value like we do with maps and arrays. We could consider adding something like `my_tuple.0`, but the index will have to be known at compile time.

Reply:
Ah, interesting! I didn't know that this is how rust handles it but it makes sense now that I think about it. I'm onboard :+1: